pkg/transforms/mqttsecret_test.go

```diff
@@ -0,0 +1,388 @@
+kX0EMnYch0e7urs9o1M6JWJGlWZQxgVnxekbFDPfRelR1m0zFnbfXG2rnfuRpVEL
+6SGxFU8+v1VepAHLvhS2VULYbWBOHZsh1yCteUXdePMYIN7c71qaCyC89N3GBia5
+uXOR
+-----END CERTIFICATE-----
+`
+const testclientkey = `-----BEGIN RSA PRIVATE KEY-----
+MIIEpAIBAAKCAQEA4TlobJoFgNoCc5Znb0OzVoMypoay1RSTAhnU0arpHVugUMZM
+O6oxSt371MN+e4cUxoes4uhNqeVG7AxUkdMCNJbzjAmJeDQtLKYHcY4YI30HHWCW
+0c8SxEsrj6DzjizgKZcUdX4H6HwAltOp/RZYJTBVVexE1WYOheTNJuw5QeNbTGpf
+pKM7RuHADnytLbrSiK09FZYx23PIsLhx8b7+k1AtRFGhFqDRMF6Fqbo6xdU8hZ1e
+AvJP5t87U/PWeQ9ld2lxd3fQxiP4IBQs1QI2gTp5O41ifRCpO7scXRaFweyPAgMV
+OQ42eVZiJUR37AF/nVzXxB5NiTH9Ij/c/shJvQIDAQABAoIBADPL4BgZ0+ouOSIc
+FO2hxDzBL4TctYQLl0OEbU1K4RG/YL8y25VdLrjpFGF6FDyUdFK0IS6N/k50TDs9
+GrXusTMnBBvQlazvUvRRuqSC6UpAFsLK0+SsmsRKBVqiyWCJMYRfGnVq5qaw3fHR
+++YYnWzwELASBkKNlgl09TleWkysbnZbWIMQ5Qm0k+s/9vvjooA2aMXTeLtyhGfI
+49OvyCrrX5v7ILdHl7RGAyPRT+ipyt1i0fAqHk4ouLdTRrAx4S5TvUpszrts1P8f
+5ggLd1s6RVTz27uASu3U/gLH630m1PU46d02UI1tWen3TgRm/VqjO2aqkZaZispQ
+HwTRZIECgYEA9rL7KoZflVQJ4ndg3V522BhAciN99taYWHr018kG5vNVGFBHSVOt
+De0gb7z8FhK0Zs4MifU3b03qr7Ac1+p0zIAwATPT4TOLzc4SKBd33TZk/JCZCGSR
+hqQPF0FZ+EKJqh7yif+ssFXp0xKrNybm58Z7jfF8vWMdz0QkJ1pZkn8CgYEA6bcp
+YkH6IoHmCZ5hWE3/hYQcvfcM10z0cWTTKstxgSid9dj0HUqxMsFhBF1yzUtsDZQB
+E933gZyj/LE5Z/EbqUSX0H/M0P7Uwtj9lS7W/vQdOQMfAciqggNKhyaBnBYsxw9l
+5IelOxGF+taEvDkPsVt9cvZm/nbf+irU5JLCzcMCgYEA8o3/jUwY5oV+QoAFaSHb
+z5PoqVBkJTHREA20dgVdF+3fmMw1is8Os0aWQcaaREmXvgyRH4NOQc1mFd8ePNx0
+giz3BfejNySrLGqUR37rh0BYAktZa3sV6j+b5s2GXCVvnShYZ35OmAGgqLsORGen
+V/M6v9DTSJIPWR4yPc8DipkCgYEAhmtW/PFPaRtm7+9Ms5ogtWz3jvaRRx82lCVW
+Io3iGVQADc8bD+HOqo94Oid5CMQxQFn4iLGoUb6Cvqo7hyGwNBmEa2GlripyuiJN
+LslC1F4YlJrL8Z21G5PDAJpP/zLtzAt6Igc2LBP3B/7rVspG0U36h+1Z7U73oQ2T
+ZmdWbTsCgYALxjB0NvqBk+TNYMZFysqZnI3CxYQXwHfElQQQUqcQnunAOLJ8H+nb
+JryGx90ylYY2Mh2U273435uwQcX1g5gu3rBF8McHKj5EYSVDgpeBMx8ej2ENvW7q
+CR6KVnoNdMwJZM3ARpBYNlhFTzDyew2WYLitZsN/uV8t+XxJFDyJQA==
+-----END RSA PRIVATE KEY-----
+`
+
+func TestMQTTValidateSecrets(t *testing.T) {
+	sender := NewMQTTSecretSender(MQTTSecretConfig{}, false)
+	tests := []struct {
+		Name             string
+		AuthMode         string
+		secrets          mqttSecrets
+		ErrorExpectation bool
+		ErrorMessage     string
+	}{
+		{"No Auth No error", AuthModeNone, mqttSecrets{}, false, ""},
+		{"UsernamePassword No Error", AuthModeUsernamePassword, mqttSecrets{
+			username: "user",
+			password: "password",
+		}, false, "AuthModeUsernamePassword selected however username or password was not found at secret path"},
+		{"UsernamePassword Error no username", AuthModeUsernamePassword, mqttSecrets{
+			password: "password",
+		}, true, "AuthModeUsernamePassword selected however username or password was not found at secret path"},
+		{"UsernamePassword Error no password", AuthModeUsernamePassword, mqttSecrets{
+			username: "user",
+		}, true, "AuthModeUsernamePassword selected however username or password was not found at secret path"},
+		{"ClientCert No Error", AuthModeCert, mqttSecrets{
+			certpemblock: []byte("----"),
```

Comment:
why "----"? placeholder that should be changed to using above cert/key?

Reply:
wanted to highlight that a valid cert is not needed here - only checking that something exists